### PR TITLE
Dsb tagging support

### PR DESCRIPTION
**Pull request by Artlvns**
chore: remove .vscode from .gitignore.
chore: add vscode common workspace config.
feat: Introduced DSB mandatory resourcegroup tag costCenter with default 142 - costCenter for IKT. .
chore: TODO's for systemId mandatory tags prepared to be applied when systemId's are known
style: README.md linting and spelling.
docs: Update README TFdocs with new variable.
style: TFlint.
chore: bump azurerm to 4.3.0

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -53,24 +62,14 @@ variable "azure_region" {
     condition     = length(var.azure_region) > 0
   }
 }
-variable "created_by_tag" {
-  description = "Tag to use when naming resources."
+variable "costcenter_tag" {
```
Comment (Laffs2k5) [inline variables.tf]: This name is confusing. This is not the tag, it's the tag value. At least change the description.

File: variables.tf [hunk 2]
```diff
@@ -140,3 +138,20 @@ variable "network_rules" {
...
+  type        = string
+  nullable    = false
+  default     = "terraform-remote-backend-state"
+  validation {
+    error_message = "The 'state_container_name' must be supplied and cannot be null or empty string."
+    condition     = length(var.state_container_name) > 0
+  }
+}
+#TODO: Uncomment the following lines when the systemId tag is available and add default value.
+#variable "systemid_tag" {
```
Comment (Laffs2k5) [inline variables.tf]: Same confusing name/description here.

**Review by Laffs2k5 — APPROVED** [review]
This calls for a new minor release from main IMO. Which results in a new `1.x` release.

**Note:** remember to also backport changes to the [v0 branch](https://github.com/dsb-norge/tf-mod-azure-terraform-state-container/tree/v0-branch) with a separate PR and a minor release of `v0.x`

File: variables.tf (hunk 1 above)
Comment (Artlvns) [reply variables.tf]: I'll refactor variable names. I see the same inconsistency for "created_by_tag" variable in this module.

**Pull request merged**